ddtrace/tracer/rules_sampler.go

```diff
@@ -199,17 +199,18 @@ func SpanNameServiceMPSRule(name, service string, rate, limit float64) SamplingR
 // Its value is the number of spans to sample per second.
 // Spans that matched the rules but exceeded the rate limit are not sampled.
 type traceRulesSampler struct {
+	m          sync.RWMutex
 	rules      []SamplingRule // the rules to match spans with
 	globalRate float64        // a rate to apply when no rules match a span
 	limiter    *rateLimiter   // used to limit the volume of spans sampled
 }
 
 // newTraceRulesSampler configures a *traceRulesSampler instance using the given set of rules.
 // Invalid rules or environment variable values are tolerated, by logging warnings and then ignoring them.
-func newTraceRulesSampler(rules []SamplingRule) *traceRulesSampler {
+func newTraceRulesSampler(rules []SamplingRule, traceSampleRate float64) *traceRulesSampler {
 	return &traceRulesSampler{
 		rules:      rules,
-		globalRate: globalSampleRate(),
+		globalRate: traceSampleRate,
```

Comment:
If we don't plan to use `globalSampleRate()` anywhere but in `newUnstartedTracer`, should we still keep it here as is?

Reply:
This has been discussed offline but the tl;dr we kept it here for consistency as the sampler parses all its configs in this file. Also the tracer.go file doesn't contain any helpers to parse configs.